src/settings/PathConfigStorage.cpp

```diff
@@ -163,6 +222,65 @@ PathConfigStorage::PathConfigStorage(const SliceDataStorage& storage, int layer_
     }
 }
 
+const GCodePathConfig *PathConfigStorage::MeshPathConfigs::getInset0Config(int layer_nr) const
+{
+    return (layer_nr == 0)? &inset0_config_layer0 : &inset0_config;
+}
+
+const GCodePathConfig *PathConfigStorage::MeshPathConfigs::getInsetXConfig(int layer_nr) const
+{
+    return (layer_nr == 0)? &insetX_config_layer0 : &insetX_config;
+}
+
+const GCodePathConfig *PathConfigStorage::MeshPathConfigs::getSkinConfig(int layer_nr) const
+{
+    return (layer_nr == 0)? &skin_config_layer0 : &skin_config;
+}
+
+const GCodePathConfig *PathConfigStorage::MeshPathConfigs::getPerimeterGapConfig(const int layer_nr) const
+{
+    return (layer_nr == 0)? &perimeter_gap_config_layer0 : &perimeter_gap_config;
+}
+
+const GCodePathConfig *PathConfigStorage::MeshPathConfigs::getInfillConfig(const int layer_nr, const int combine_count) const
+{
+    return (layer_nr == 0)? &infill_config_layer0 : &infill_config[combine_count];
+}
+
+const GCodePathConfig *PathConfigStorage::getPrimeTowerConfig(const int layer_nr, const int extruder_nr) const
+{
+    return (layer_nr == 0)? &prime_tower_config_per_extruder_layer0[extruder_nr] : &prime_tower_config_per_extruder[extruder_nr];
+}
+
+const GCodePathConfig *PathConfigStorage::getSupportInfillConfig(const int layer_nr) const
+{
+    return (layer_nr == 0)? &support_infill_config_layer0 : &support_infill_config;
+}
+
+const GCodePathConfig *PathConfigStorage::getSupportRoofConfig(const int layer_nr) const
+{
+    return (layer_nr == 0)? &support_roof_config_layer0 : &support_roof_config;
+}
+
+void PathConfigStorage::MeshPathConfigs::smoothAllSpeeds(GCodePathConfig::SpeedDerivatives first_layer_config, int layer_nr, int max_speed_layer)
```

Comment:
Why did you put these in a separate function? It seemed logical to have them in `handleInitialLayerSpeedup`.

(Not blocking.)